lang/java/reef-wake/wake/src/test/java/org/apache/reef/wake/test/remote/TransportHttpTest.java

```diff
@@ -0,0 +1,162 @@
+import org.junit.rules.TestName;
+
+import java.net.InetSocketAddress;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.logging.Level;
+
+
+/**
+ * Tests for Transport via http.
+ */
+public class TransportHttpTest {
+  private final LocalAddressProvider localAddressProvider;
+  private final TransportFactory tpFactory;
+
+  public TransportHttpTest() throws InjectionException {
+    final Injector injector = Tang.Factory.getTang().newInjector();
+    this.localAddressProvider = injector.getInstance(LocalAddressProvider.class);
+    this.tpFactory = injector.getInstance(MessagingTransportFactory.class);
+  }
+
+  private static final String LOG_PREFIX = "TEST ";
+  @Rule
+  public TestName name = new TestName();
+
+  @Test
+  public void testHttpTransportString() throws Exception {
+    System.out.println(LOG_PREFIX + name.getMethodName());
+    LoggingUtils.setLoggingLevel(Level.INFO);
+
+    final Monitor monitor = new Monitor();
+    final TimerStage timer = new TimerStage(new TimeoutHandler(monitor), 2000, 2000);
+
+    final int expected = 2;
+
+    final String hostAddress = localAddressProvider.getLocalAddress();
+
+    // Codec<String>
+    final ReceiverStage<String> stage =
+        new ReceiverStage<>(new ObjectSerializableCodec<String>(), monitor, expected);
+    final Transport transport = tpFactory.newInstance(hostAddress, 0, stage, stage, 1, 10000, 101);
+    final int port = transport.getListeningPort();
+
+    // sending side
+    final Link<String> link = transport.open(
+        new InetSocketAddress(hostAddress, port),
+        new ObjectSerializableCodec<String>(),
+        new LoggingLinkListener<String>());
+    link.write(new String("hello1"));
+    link.write(new String("hello2"));
+
+    monitor.mwait();
+    transport.close();
+    timer.close();
+
+    Assert.assertEquals(expected, stage.getCount());
+  }
+
+  @Test
+  public void testHttpTransportTestEvent() throws Exception {
+    System.out.println(LOG_PREFIX + name.getMethodName());
+    LoggingUtils.setLoggingLevel(Level.INFO);
+
+    final Monitor monitor = new Monitor();
+    final TimerStage timer = new TimerStage(new TimeoutHandler(monitor), 2000, 2000);
+
+    final int expected = 2;
+    final String hostAddress = localAddressProvider.getLocalAddress();
+
+    // Codec<TestEvent>
+    final ReceiverStage<TestEvent> stage =
+        new ReceiverStage<>(new ObjectSerializableCodec<TestEvent>(), monitor, expected);
+    final Transport transport = tpFactory.newInstance(hostAddress, 0, stage, stage, 1, 10000, 101);
+    final int port = transport.getListeningPort();
+
+    // sending side
+    final Link<TestEvent> link = transport.open(
+        new InetSocketAddress(hostAddress, port),
+        new ObjectSerializableCodec<TestEvent>(),
+        new LoggingLinkListener<TestEvent>());
+    link.write(new TestEvent("hello1", 0.0));
+    link.write(new TestEvent("hello2", 1.0));
+
+    monitor.mwait();
+    transport.close();
```

Comment:
`Transport` implements `AutoCloseable` - can use it in `try` block instead of calling `.close()` explicitly